src/core/io/platform/f4xx/ADCf4xx.cpp

```diff
@@ -84,109 +127,222 @@ void ADCf4xx::initADC(uint8_t num_channels) {
 
+    // Combines the ADC channel with the ADC peripherals it supports, to avoid having multi-layered switch statements
     switch (pin) {
     case Pin::PA_0:
-        adcChannel.Channel = ADC_CHANNEL_0;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_0);
         break;
     case Pin::PA_1:
-        adcChannel.Channel = ADC_CHANNEL_1;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_1);
         break;
     case Pin::PA_2:
-        adcChannel.Channel = ADC_CHANNEL_2;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_2);
         break;
     case Pin::PA_3:
-        adcChannel.Channel = ADC_CHANNEL_3;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_3);
         break;
     case Pin::PA_4:
-        adcChannel.Channel = ADC_CHANNEL_4;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_4);
         break;
     case Pin::PA_5:
-        adcChannel.Channel = ADC_CHANNEL_5;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_5);
         break;
     case Pin::PA_6:
-        adcChannel.Channel = ADC_CHANNEL_6;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_6);
         break;
     case Pin::PA_7:
-        adcChannel.Channel = ADC_CHANNEL_7;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_7);
         break;
     case Pin::PB_0:
-        adcChannel.Channel = ADC_CHANNEL_8;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_8);
         break;
     case Pin::PB_1:
-        adcChannel.Channel = ADC_CHANNEL_9;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_9);
         break;
     case Pin::PC_0:
-        adcChannel.Channel = ADC_CHANNEL_10;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_10);
         break;
     case Pin::PC_1:
-        adcChannel.Channel = ADC_CHANNEL_11;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_11);
         break;
     case Pin::PC_2:
-        adcChannel.Channel = ADC_CHANNEL_12;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_12);
         break;
     case Pin::PC_3:
-        adcChannel.Channel = ADC_CHANNEL_13;
+        channel = CHANNEL_SET(1, 1, 1, ADC_CHANNEL_13);
         break;
     case Pin::PC_4:
-        adcChannel.Channel = ADC_CHANNEL_14;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_14);
         break;
     case Pin::PC_5:
-        adcChannel.Channel = ADC_CHANNEL_15;
+        channel = CHANNEL_SET(1, 1, 0, ADC_CHANNEL_15);
         break;
     default:
+        channel = 0;
         break; // Should never get here
     }
 
+    // This checks if the pin being used supports the ADC being used
+    if (checkSupport(adcPeriph, channel)) {
+        // Masks channel back to proper value (Zero's out ADC information bits)
+        adcChannel.Channel = channel & 0x1F;
+    } else {
+        log::LOGGER.log(log::Logger::LogLevel::ERROR, "ADC %d DOES NOT SUPPORT PIN 0x%x!!", (adcNum + 1), pin);
+        // Causes HARD FAULT if pin does not support the ADC peripheral being used. THIS IS INTENTIONAL!
+        *((volatile int*) 0xFFFFFFFF) = 0; // This address is invalid
```

Comment:
Ok so the problem is that if I dont hard fault, the programmer (without logger enabled) won't be able to tell why the adc is just reading 0's, as it wont break or anything, it will keep going as if everything worked. 
So should i just like, log it and let it run? Whats ur suggestion

Reply:
I think you should log at error level and let it go. If you turn the logger off, that's your problem. @mjh9585 what do you think?